workitem/link/link_repository_blackbox_test.go

```diff
@@ -40,296 +26,169 @@ func TestRunLinkRepoBlackBoxTest(t *testing.T) {
-	require.Nil(s.T(), err)
-
-	_, err = s.workitemLinkRepo.Create(s.Ctx, s.parent1.ID, child3.ID, s.testTreeLinkTypeID, s.testIdentity.ID)
-	require.Nil(s.T(), err)
-
-	offset := 0
-	limit := 1
-	res, count, err := s.workitemLinkRepo.ListWorkItemChildren(s.Ctx, s.parent1.ID, &offset, &limit)
-	require.Nil(s.T(), err)
-	require.Len(s.T(), res, 1)
-	require.Equal(s.T(), 3, int(count))
+func (s *linkRepoBlackBoxTest) TestList() {
+	// tests total number of workitem children returned by list is equal to the
+	// total number of workitem children created and total number of workitem
+	// children in a page are equal to the "limit" specified
+	s.T().Run("ok - count child work items", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.WorkItems(4), // parent + child 1-3
+			tf.WorkItemLinkTypes(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinkTypes[idx].ForwardName = "parent of"
+				return nil
+			}),
+			tf.WorkItemLinks(3, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinks[idx].SourceID = fxt.WorkItems[0].ID
+				fxt.WorkItemLinks[idx].TargetID = fxt.WorkItems[idx+1].ID
+				return nil
+			}),
+		)
+
+		offset := 0
+		limit := 1
+		res, count, err := s.workitemLinkRepo.ListWorkItemChildren(s.Ctx, fxt.WorkItems[0].ID, &offset, &limit)
+		require.Nil(t, err)
+		require.Len(t, res, 1)
+		require.Equal(t, 3, int(count))
+	})
 }
 
-func (s *linkRepoBlackBoxTest) TestWorkItemHasNoChildAfterDeletion() {
-	// given
-	// create a work item link...
-	s.T().Log(fmt.Sprintf("creating link with treelinktype.ID=%v", s.testTreeLinkTypeID))
-	wil, err := s.workitemLinkRepo.Create(s.Ctx, s.parent1.ID, s.child.ID, s.testTreeLinkTypeID, s.testIdentity.ID)
-	require.Nil(s.T(), err)
-	// ... and remove it
-	err = s.workitemLinkRepo.Delete(s.Ctx, wil.ID, s.testIdentity.ID)
-	require.Nil(s.T(), err)
-
-	// when
-	hasChildren, err := s.workitemLinkRepo.WorkItemHasChildren(s.Ctx, s.parent1.ID)
-	// then
-	assert.Nil(s.T(), err)
-	assert.False(s.T(), hasChildren)
+func (s *linkRepoBlackBoxTest) TestWorkItemHasChildren() {
+	s.T().Run("work item has no child after deletion", func(t *testing.T) {
+		// given a work item link
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.WorkItems(2), // parent + child 1
+			tf.WorkItemLinkTypes(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinkTypes[idx].ForwardName = "parent of"
+				return nil
+			}),
+			tf.WorkItemLinks(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemLinks[idx].SourceID = fxt.WorkItems[0].ID
+				fxt.WorkItemLinks[idx].TargetID = fxt.WorkItems[idx+1].ID
+				return nil
+			}),
+		)
+
+		// when this work item link is deleted
+		err := s.workitemLinkRepo.Delete(s.Ctx, fxt.WorkItemLinks[0].ID, fxt.Identities[0].ID)
+		require.Nil(t, err)
+
+		// then it must not have any child
+		hasChildren, err := s.workitemLinkRepo.WorkItemHasChildren(s.Ctx, fxt.WorkItems[0].ID)
+		// then
+		assert.Nil(t, err)
```

Comment:
it could be `require.Nil(t, err)` here, to be consistent with all other error checkings.

Reply:
will do

Reply:
fixed in 99e4ecc